src/index.js

```diff
@@ -1,29 +1,78 @@
 import './style.css';
 
+const form = document.querySelector('form');
+const input = document.querySelector('.input');
 const todoContainer = document.querySelector('.todoContainer');
 
-const lists = [{
-  description: 'Work hard on coding',
-  bool: 'false',
-  index: '1',
-},
-{
-  description: 'Be smart and consistent',
-  bool: 'false',
-  index: '2',
-}, {
-  description: 'You can do it James ',
-  bool: 'false',
-  index: '3',
-}];
+let todo;
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+const store = () => {
+  todo = {
+    Description: input.value,
+    id: todos.length,
+    completed: false,
+  };
+  todos.push(todo);
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
 
-lists.forEach((task) => {
-  const ul = document.createElement('ul');
+const clear = () => {
+  input.value = '';
+};
+
+const removeBook = (id) => {
+  todos = todos.filter((books) => books.id !== id);
+  todos.forEach((todo, id) => {
+    todo.id = id;
+  });
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
+
+const addTask = (todo) => {
+  const ul = document.createElement('div');
+  const checkBox = document.createElement('input');
+  checkBox.type = 'checkbox';
+  checkBox.classList.add('checkBox');
+  const newInp = document.createElement('input');
+  newInp.type = 'text';
+  newInp.classList.add('newInput');
+  newInp.value = todo.Description;
+  const icon = document.createElement('i');
+  icon.classList.add('fa-solid');
+  icon.classList.add('fa-ellipsis-vertical');
+  icon.classList.add('dots');
   const hr = document.createElement('hr');
-  ul.innerHTML = `
-    <li>
-    <input class="checkBox" type="checkbox">${task.description}<i class="fa-solid fa-ellipsis-vertical dots"></i>
-    </li>
-    `;
-  todoContainer.append(ul, hr);
+  ul.append(checkBox, newInp, icon, hr);
+  todoContainer.append(ul);
+  icon.addEventListener('click', () => {
+    icon.parentElement.remove();
+    removeBook(todo.id);
+  });
+};
+todos.forEach(addTask);
+
+const editTodoList = () => {
+  const editInput = document.querySelectorAll('.newInput');
+  editInput.forEach((edits, indexy) => {
+    edits.addEventListener('change', () => {
+      todos.forEach((todo, index) => {
+        if (indexy === index) {
+          todo.Description = edits.value;
+          localStorage.setItem('todos', JSON.stringify(todos));
+        }
+      });
+    });
+  });
+};
+editTodoList();
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  if (input.value !== '') {
+    store();
+    addTask(todo);
+    clear();
+  } else {
+    alert('Please enter a title and author');
+  }
```

Comment:
- [ ] The property index should be set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal 5) please address that

Reply:
Hi@DeliceLydia,
Thanks for the review , regarding the required changes requested, i have actually tested the code and it runs pecfectly through the iteration. I will recommend you run it using npm start to confirm what i am saying. Secondly when you run